usage/test-framework/src/test/java/org/apache/brooklyn/test/framework/TestSensorTest.java

```diff
@@ -0,0 +1,270 @@
+        final String sensorValue = String.format("%s%s%s", UUID.randomUUID().toString(), time, UUID.randomUUID().toString());
+
+        //Add Sensor Test for STRING sensor
+        app.createAndManageChild(EntitySpec.create(TestSensor.class)
+                .configure(TestSensor.TARGET_ENTITY, app)
+                .configure(TestSensor.SENSOR_NAME, STRING_SENSOR.getName())
+                .configure(TestSensor.ASSERTIONS, ImmutableMap.of("regex", String.format(".*%s.*", time))));
+        app.createAndManageChild(EntitySpec.create(TestSensor.class)
+                .configure(TestSensor.TARGET_ENTITY, app)
+                .configure(TestSensor.SENSOR_NAME, BOOLEAN_SENSOR.getName())
+                .configure(TestSensor.ASSERTIONS, ImmutableMap.of("regex", "true")));
+
+        //Set STRING sensor
+        app.sensors().set(STRING_SENSOR, sensorValue);
+        app.sensors().set(BOOLEAN_SENSOR, true);
+
+
+        app.start(ImmutableList.of(loc));
+    }
+
+    @Test
+    public void testAssertRegexFail() {
+        boolean sensorTestFail = false;
+        final String sensorValue = String.format("%s%s%s", UUID.randomUUID().toString(), System.currentTimeMillis(), UUID.randomUUID().toString());
+
+        //Add Sensor Test for STRING sensor
+        app.createAndManageChild(EntitySpec.create(TestSensor.class)
+                .configure(TestSensor.TARGET_ENTITY, app)
+                .configure(TestSensor.SENSOR_NAME, STRING_SENSOR.getName())
+                .configure(TestSensor.ASSERTIONS, ImmutableMap.of("regex", String.format(".*%s.*", UUID.randomUUID().toString()))));
+
+        //Set STRING sensor
+        app.sensors().set(STRING_SENSOR, sensorValue);
+
+
+        try {
+            app.start(ImmutableList.of(loc));
+        } catch (final PropagatedRuntimeException pre) {
+            final AssertionError assertionError = Exceptions.getFirstThrowableOfType(pre, AssertionError.class);
+            assertThat(assertionError).isNotNull().as("An assertion error should have been thrown");
+            sensorTestFail = true;
+        } finally {
+            assertThat(sensorTestFail).isTrue().as("regex assert should have failed");
+        }
+    }
+
+    @Test
+    public void testAssertRegexOnNullSensor() {
+        boolean sensorTestFail = false;
+        final String sensorValue = String.format("%s%s%s", UUID.randomUUID().toString(), System.currentTimeMillis(), UUID.randomUUID().toString());
+
+        //Add Sensor Test for STRING sensor
+        app.createAndManageChild(EntitySpec.create(TestSensor.class)
+                .configure(TestSensor.TARGET_ENTITY, app)
+                .configure(TestSensor.SENSOR_NAME, STRING_SENSOR.getName())
+                .configure(TestSensor.ASSERTIONS, ImmutableMap.of("regex", String.format(".*%s.*", UUID.randomUUID().toString()))));
+
+        try {
+            app.start(ImmutableList.of(loc));
+        } catch (final PropagatedRuntimeException pre) {
```

Comment:
I've added a few things to `Asserts` to simplify this; you should be able to simplify many of the tests here using:

    try {
        app.start(ImmutableList.of(loc));
        Asserts.shouldHaveFailedPreviously();
    } catch (Throwable t) {
        Asserts.expectedFailureOfType(t, AssertionError.class);
        // and assert the error message has good detail, including the actual and expected values
        // (assuming we set `regexExpected` as ".*"+UUID.randomUUID()+".*" and pass that in above
        Asserts.expectedFailureContainsIgnoreCase(t, "sensor", "regex", /* actual */ "null", /* expected */ regexExpected);
    }